qhana_plugin_runner/db/models/tasks.py

```diff
@@ -95,19 +95,17 @@ class ProcessingTask:
     Attributes:
         id (int, optional): automatically generated database id. Use the id to fetch this information from the database.
         task_name (str): the name of the (logical) task corresponding to this information object
-        task_id (Optional[str], optional): the final celery task id to wait for to declare this task finished. If not supplied this task will never be marked as finished. In multi-step plugins, this attribute is also used for intermediate task results. Only needed to check task status of :py:func:`qhana-plugin-runner.qhana_plugin_runner.tasks.save_task_result`.
         started_at (datetime, optional): the moment the task was scheduled. (default :py:func:`~datetime.datetime.utcnow`)
         finished_at (Optional[datetime], optional): the moment the task finished successfully or with an error.
         parameters (str): the parameters for the task. Task parameters should already be prepared and error checked before starting the task.
         data (dict): dict-like key-value store for additional lightweight task data. New elements of type :class:`TaskData` can be added or retrieved as in a dict using ``key`` as key.
-        multi_step (bool): set to ``True`` if task data is used for a multi-step plugin.
         steps (OrderingList[Step]): ordered list of steps of type :class:`Step`. Index ``number`` automatically increases when new elements are appended. Note: only use :meth:`add_next_step` to add a new step. Steps must not be deleted.
         current_step (int): index of last added step.
-        progress_value (int): progress value in multi-step plugins.
-        progress_start (int): progress start value in multi-step plugins.
-        progress_target (int): progress target value in multi-step plugins.
-        progress_unit (str): progress unit in multi-step plugins (default: "%").
-        finished_status (Optional[str], optional): the status string with witch the celery task with the ``task_id`` finished. If set then ``task_id`` may not be checked.
+        progress_value (int): progress value. ``None`` by default.
+        progress_start (int): progress start value.
+        progress_target (int): progress target value.
+        progress_unit (str): progress unit (default: "%").
+        finished_status (Optional[str], optional): the status string of the plugin execution
```

Comment:
We should probably rename this now to task status as we are refactoring this class anyway. And in the docstring we should limit this to PENDING, SUCCESS and ERROR to avoid custom status strings. Maybe even limit this with code guards.